src/Uri.php

```diff
@@ -548,6 +548,21 @@ private function filterScheme($scheme)
     }
 
     /**
+     * Filters a part of user info in a URI to ensure it is properly encoded.
+     *
+     * @param string $part
+     * @return string
+     */
+    private function filterUserInfoPart($part)
```

Comment:
@weierophinney, see my prior comment for argument in favor of using SPL `urlencode` function instead of this.